lib/Dialect/Torch/Transforms/RefineTypes.cpp

```diff
@@ -1509,9 +1514,13 @@ namespace {
 class RefineTypesPass : public RefineTypesBase<RefineTypesPass> {
   void runOnOperation() override {
     auto func = getOperation();
-    TypeAnalyzer analyzer(&getContext());
-    analyzer.run(func);
-    optimize(func, analyzer);
+    DataFlowSolver solver;
+    solver.load<dataflow::DeadCodeAnalysis>();
+    solver.load<dataflow::SparseConstantPropagation>();
+    TypeAnalyzer *analyzer = solver.load<TypeAnalyzer>();
+    if (failed(solver.initializeAndRun(func)))
+      return signalPassFailure();
+    optimize(func, *analyzer);
```

Comment:
Again, to be consistent with mlir, this should take the `solver` instead of the `analyzer` and then you can remove `getLatticeElementPublic` and instead use `lookupState`, right?

Reply:
getLatticeElementPublic is mainly for the free functions that TypeAnalysis calls. I changed the other code though.